portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,41 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    // Initialize the message list with harcoded messages
+    ArrayList<String> messageList = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    messageList.add("Hello Naa'Oli!"); 
+    messageList.add("How are you doing Naa'Oli?");
+    messageList.add("Good to see you again Naa'Oli.");
+    String message = convertToJson();
+
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Naa'Oli!</h1>");
+    response.getWriter().println(message);
+  }
+
+  public String convertToJson() {
```

Comment:
If you ever find yourself manually adding brackets and quotes, I highly recommend taking a step back and finding out if there is an existing solution. It's very difficult to get escaping correct (I think you missed sanitizing `[` characters in the message input for example).

In this case, I think others have been using the `com.googlecode.json-simple` library: https://mkyong.com/java/json-simple-example-read-and-write-json/